### PR TITLE
PLT-6881 Utxo indexer returns all spent transaction outputs when not specifying `unspentBeforeSlotNo` field

### PR DESCRIPTION
**Pull request by berewt**
Fix UTXO bug and modify the existing tests to catch regression

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting main unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [ ] Reviewer requested

### CODE REVIEW

File: marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs [hunk 1]
```diff
@@ -164,17 +164,18 @@ propAllQueryUtxosShouldBeUnspent = Hedgehog.property $ do
...
-  let upperBound = maximum $ fmap (Utxo._blockInfoSlotNo . ueBlockInfo) events
-      -- we want to query for all addresses
-      addressQueries :: [StorableQuery Utxo.UtxoHandle] =
+  -- we want to query for all addresses
+  let addressQueries :: [StorableQuery Utxo.UtxoHandle] =
         List.nub
           . fmap
             ( Utxo.QueryUtxoByAddressWrapper
-                . flip Utxo.QueryUtxoByAddress (Utxo.lessThanOrEqual upperBound) --  maxBound will overflow. See TODO below
+                . flip Utxo.QueryUtxoByAddress (Utxo.Interval Nothing upperBound) --  maxBound will overflow. See TODO below
```
Comment (koslambrou) [inline marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs:178]: Why the change? You can use `lessThanOrEqual` here no?

**Review by koslambrou — APPROVED** [review]

File: marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs (hunk 1 above)
Comment (berewt) [reply marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs:178]: We had a `SlotNo` as an `upperBound`, we now have a `Maybe SlotNo` (we didn't handle the `Nothing` case before, which would have caught the issue.) And `lessThanOrEqual` only works with a `SlotNo`.
